pkg/kubelet/network/hostport/hostport.go

```diff
@@ -128,27 +136,28 @@ func (h *handler) openHostports(pod *v1.Pod) error {
 	return nil
 }
 
+func getPodFullName(pod *PodPortMapping) string {
+	// Use underscore as the delimiter because it is not allowed in pod name
+	// (DNS subdomain format), while allowed in the container name format.
+	return pod.Name + "_" + pod.Namespace
+}
+
 // gatherAllHostports returns all hostports that should be presented on node,
 // given the list of pods running on that node and ignoring host network
 // pods (which don't need hostport <-> container port mapping).
-func gatherAllHostports(activePods []*ActivePod) (map[v1.ContainerPort]targetPod, error) {
-	podHostportMap := make(map[v1.ContainerPort]targetPod)
-	for _, r := range activePods {
-		if r.IP.To4() == nil {
-			return nil, fmt.Errorf("Invalid or missing pod %s IP", kubecontainer.GetPodFullName(r.Pod))
+func gatherAllHostports(activePortMapping []*PodPortMapping) (map[*PortMapping]targetPod, error) {
```

Comment:
Modify `activePortMapping ` to `activePodPortMapping ` to be clearer? Other similar variables may also need to be modified. Just a nit:)